server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -208,15 +210,39 @@
 		"/:tenantId/document/:id/revokeToken",
 		validateRequestParams("tenantId", "id"),
 		validateTokenRevocationClaims(),
-		verifyStorageToken(tenantManager, config),
+		verifyStorageToken(tenantManager, config, tokenManager),
 		throttle(tenantThrottler, winston, tenantThrottleOptions),
 		async (request, response, next) => {
 			const documentId = getParam(request.params, "id");
 			const tenantId = getParam(request.params, "tenantId");
 			const lumberjackProperties = getLumberBaseProperties(documentId, tenantId);
 			Lumberjack.info(`Received token revocation request.`, lumberjackProperties);
```

Comment:
>request.body.jti; [](http://example.com/codeflow?start=19&length=17)

Should we pass JTI from the request body or a token claim?

Reply:
I had a chat with Manish about it before and I don't have a preference for one over another. If we want to put it in JWT claim, we need a new custom field other than JTI since it is used to auth the request itself. My thinking is that JWT has all the information to authenticate the request while the body has the information to fulfill the request. Please let me know if you feel it is more appropriate to put in in JWT claims. Thanks

Reply:
I think JTI should be from the request body. JWT claim should be generic which tells us whether a user has permission to revoke a token a not. The actual JTI  should not be within the JWT itself. 
If we take a usual case, let's say we have a user management system. If an admin wants to delete a user, he sends the userId to delete in the request (query string or request body). The application authenticates the request and authorizes it (the request initiator has permission to delete the user) and only then deletes the user.

Reply:
The reason why I was thinking passing JTI as a token claim is to enable the scenario where c1 delegates the revocation to c2, since c2 can't be fully trusted by c1 and c2 could try to revoke a different token if the JTI is passed in the body. We intend c1 to revoke the token, so passing JTI in body  is fine. In the future, if we find we need to enable the delegation scenario, we can pass JTI as a claim. Fortunately, it won't be a breaking change.